pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationProviderToken.java

```diff
@@ -126,9 +137,40 @@ private static String validateToken(final String token) throws AuthenticationExc
     @SuppressWarnings("unchecked")
     private Jwt<?, Claims> authenticateToken(final String token) throws AuthenticationException {
         try {
-            return Jwts.parser()
+            Jwt<?, Claims> jwt = Jwts.parser()
                     .setSigningKey(validationKey)
                     .parse(token);
+
+            if (audienceClaim != null) {
+                if (audience == null){
+                    throw new JwtException("Token Audience Claim [" + audienceClaim
+                                           + "] configured, but Audience stands for this broker not.");
+                }
+
+                Object object = jwt.getBody().get(audienceClaim);
+                if (object == null) {
+                    throw new JwtException("Found null Audience in token, for claimed field: " + audienceClaim);
```

Comment:
If the broker configures the audience claim and receives a token that doesn't contain the audience claim, we should throw AuthenticationException.

Reply:
Right. currently, the JwtException will be wrapped into a AuthenticationException in line 174 below.